README.md

```diff
@@ -2,73 +2,250 @@
+  - [Customization](#customization)
 - [Development Workflow](#development-workflow)
-  - [Requirements](#requirements)
-  - [Install](#install)
-  - [Build](#build)
-  - [Serve](#serve)
-  - [Test](#test)
-  - [Release](#release)
-- [Contributing](#contributing)
 - [Credits](#credits)
 
-<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+## Installation and Usage
+
+### Before using the library
+
+To integrate a search bar to your documentation and use this library, you need to scrap your website contents and to push it into a MeiliSearch instance.
+
+Hopefully, we provide all the tools needed and [guide you into these different steps](https://docs.meilisearch.com/resources/howtos/search_bar_for_docs.html) 🚀
+
+As a first introduction, you might only want to test this library without connecting it to your website.<br>
+You can do it by running the playground provided in this repository:
+
+```bash
+$ yarn install
+$ yarn serve
+```
+
+The data comes from MeiliSearch documentation.<br>
+Type `create an indxe` to live the MeiliSearch experience.
+
+### Basic Usage
+
+```html
+<!DOCTYPE html>
+<html>
+  <head>
+    <link
+      rel="stylesheet"
+      href="https://cdn.jsdelivr.net/npm/docs-searchbar.js@{version}/dist/cdn/docs-searchbar.min.css"
+    />
+  </head>
+
+  <body>
+    <input type="search" id="search-bar-input" />
+    <script src="https://cdn.jsdelivr.net/npm/docs-searchbar.js@{version}/dist/cdn/docs-searchbar.min.js"></script>
+    <script>
+      docsSearchBar({
+        hostUrl: "https://mymeilisearch.com",
+        apiKey: "XXX",
+        indexUid: "docs",
+        inputSelector: "#search-bar-input",
+        debug: true // Set debug to true if you want to inspect the dropdown
+      });
+    </script>
+  </body>
+</html>
+```
+
+The `hostUrl` and the `apiKey` fields are the credentials of your MeiliSearch instance.<br>
+`indexUid` is the index identifier into the scraper have pushed your website contents.<br>
+`inputSelector` is the CSS identifier of the HTML search input.
+
+_You haven't scraped your documentation contents yet? Check out [this tutorial](https://docs.meilisearch.com/resources/howtos/search_bar_for_docs.html)._
+
+**WARNING: We recommend providing the MeiliSearch public key, which is enough to perform searches.<br>
+Read more about [MeiliSearch authentication](/guides/advanced_guides/authentication.md).**
+
+### Customization
+
+The default behavior of this library fits perfectly for a documentation search bar, but you might need some customizations.
+
+- [Optional parameters](#optional-parameters-) (when calling `docsSearchBar` method)
+- [Styling](#styling-) (with CSS)
+
+#### Optional parameters <!-- omit in toc -->
+
+When calling the `docsSearchBar` method, you can add optional fields:
+
+##### `queryHook` <!-- omit in toc -->
+
+This method is called on every keystroke to transform the typed keywords before querying MeiliSearch. By default, it does not do anything.
```

Comment:
I changed to `It doesn't do anything by default.` since it is used below in `transformData`.
```suggestion
This method is called on every keystroke to transform the typed keywords before querying MeiliSearch. It doesn't do anything by default.
```